doc/devel/release_guide.rst

```diff
@@ -236,15 +239,29 @@ done for pre-releases)::
    git branch v2.0.0-doc
    git push DANGER v2.0.0-doc
 
-and if this is a major or minor release, also create a bug-fix branch (a micro
-release will be cut from this branch)::
+Update (or create) the ``v2.0-doc`` milestone.
+The description should include the instruction for meeseeksmachine to backport changes
+with the ``v2.0-doc`` milestone to both the ``v2.0.x`` branch and the ``v2.0.0-doc`` branch::
 
-   git branch v2.0.x
+   Documentation changes (.rst files and examples)
+
+   on-merge: backport to v2.0.x
+   on-merge: backport to v2.0.0-doc
+
+Check all active milestones for consistency. Older doc milestones should also backport to
+higher minor versions (e.g. ``v3.6-doc`` should backport to both ``v3.6.x`` and ``v3.7.x``
+if the ``v3.7.x`` branch exists)
 
 On this branch un-comment the globs from :ref:`release_chkdocs`.  And then ::
 
-   git push DANGER v2.0.x
+   git push DANGER v2.1.x
```

Comment:
I'm confused why you're pushing `v2.1.x` when it's only been created in the next step?

Reply:
I'm actively reworking this, I knew it was not ready (I had a discussion with Tom, which came to the same conclusions as these comments, just hadn't actually written out the updated text yet)